src/constants/MarkdownFiles/posts/2025-07-27-gsoc-25-MebinThattil-week08.md

```diff
@@ -0,0 +1,94 @@
+---
+title: "GSoC ’25 Week 08 Update by Mebin J Thattil"
+excerpt: "Platform Agnostic Inference, Profanity Filters and Kokoro Optimizations"
+category: "DEVELOPER NEWS"
+date: "2025-07-27"
+slug: "2025-07-27-gsoc-25-mebinthattil-week8"
+author: "@/constants/MarkdownFiles/authors/mebin-thattil.md"
+tags: "gsoc25,sugarlabs,week08,mebinthattil,speak_activity"
+image: "assets/Images/GSOCxSpeak.png"
+---
+
+# Week 08 Progress Report by Mebin J Thattil
+
+**Project:** [Speak Activity](https://github.com/sugarlabs/speak)  
+**Mentors:** [Chihurumnaya Ibiam](https://github.com/chimosky), [Kshitij Shah](https://github.com/kshitijdshah99)  
+**Assisting Mentors:** [Walter Bender](https://github.com/walterbender), [Devin Ulibarri](https://github.com/pikurasa)  
+**Reporting Period:** 2025-07-20 - 2025-07-27
+
+---
+
+## Goals for This Week
+
+- Optimize Kokoro + GStreamer  
+- Enable SLM inference without using compiled binaries for Llama.cpp and avoid client side binary build processes  
+- Begin dataset collection for the SugarAI-hosted LLM and gear up for fine-tuning  
+
+---
+
+## This Week’s Progress
+
+### **1. Audio Pipeline Optimizations (Kokoro & GStreamer)**
+
+After reading through the GStreamer documentation, I finally got Kokoro to stream audio directly to GStreamer, avoiding the intermediate step of writing a WAV file. This not only reduces latency, since writing to a WAV file requires the entire text to be processed, saved, and then read. While streaming we can break up the input into multiple different chunks. These chunks can be sent to Kokoro's engine for processing and then output into GStreamer’s source. This significantly reduces latency and removes the need to write to a file.
+
+I was able to get the streaming to work using the `appsrc` element. `appsrc` is a GStreamer element that lets external applications insert data into a GStreamer pipeline. The best part is that it provides external API functions.
+
+Currently, the command to construct the pipeline looks like:
+```
```

Comment:
This section is not leveraging our styling for larger chunks of code.

<img width="1134" height="396" alt="Screenshot from 2025-08-10 07-27-12" src="https://github.com/user-attachments/assets/8e284575-98ee-4e9f-8f64-d9a1a2b0bfe2" />


Please look at https://github.com/sugarlabs/www-v2/blob/f06455fe26dc69f1fe1a76e452d5879885936ab8/src/constants/MarkdownFiles/posts/TESTMarkdownFormat.md?plain=1#L48 for an example.

This is the result:

<img width="921" height="340" alt="Screenshot from 2025-08-10 07-27-50" src="https://github.com/user-attachments/assets/765be507-e303-437c-b5cf-18d59d73158a" />